app/code/Magento/AdminNotification/Block/System/Messages.php

```diff
@@ -62,15 +68,13 @@ protected function _toHtml()
     /**
      * Retrieve message list
      *
-     * @return \Magento\Framework\Notification\MessageInterface[]
+     * @return MessageInterface[]|null
      */
     public function getLastCritical()
     {
         $items = array_values($this->_messages->getItems());
-        if (isset(
-            $items[0]
-        ) && $items[0]->getSeverity() == \Magento\Framework\Notification\MessageInterface::SEVERITY_CRITICAL
-        ) {
+
+        if (isset($items[0]) && (int)$items[0]->getSeverity() === MessageInterface::SEVERITY_CRITICAL) {
             return $items[0];
         }
```

Comment:
I'm just curious if it's not better to use `!empty($items) && current($items)->getSeverity() ...` and then `return current($items)`?

Reply:
Fixed